### PR TITLE
Support custom mappers in commands

### PR DESCRIPTION
**Pull request by flash-gordon**
I added a simple realization of custom command mappers. I tested it in my project and it seems to work fine.

### CODE REVIEW

**Comment by AMHOL** [comment]
This is already supported, see https://github.com/rom-rb/rom-repository/commit/ca659158ad11dce571faf149f486c07788d3ab8f

**Comment by flash-gordon** [comment]
Oh, so I'm going to take a look tonight! Thanks!

**Comment by AMHOL** [comment]
Thanks @flash-gordon

File: lib/rom/repository.rb [hunk 1]
```diff
@@ -149,14 +149,22 @@ def command(*args, **opts, &block)
...
+          relation = name.is_a?(Symbol) ? relations[name] : name
+
           ast = relation.to_ast
           adapter = relations[relation.name].adapter
 
-          CommandCompiler[container, type, adapter, ast] >> mappers[ast]
+          if custom_mapper.nil?
+            mapper = mappers[ast]
+          else
+            mapper = relation.relation.mappers[custom_mapper]
```
Comment (solnic) [inline lib/rom/repository.rb:164]: this should be `container.mappers[custom_mapper]`

File: lib/rom/repository.rb [hunk 2]
```diff
@@ -149,14 +149,22 @@ def command(*args, **opts, &block)
...
+          custom_mapper = opts.delete(:mapper)
+          type, name = args + opts.to_a.flatten
+
+          relation = name.is_a?(Symbol) ? relations[name] : name
+
           ast = relation.to_ast
           adapter = relations[relation.name].adapter
 
-          CommandCompiler[container, type, adapter, ast] >> mappers[ast]
+          if custom_mapper.nil?
```
Comment (solnic) [inline lib/rom/repository.rb:161]: please invert this and just do `if custom_mapper ... else ... end`

**Comment by solnic** [comment]
I know this is merged but could you please look at the comments I left and clean up things in master?

File: lib/rom/repository.rb (hunk 1 above)
Comment (flash-gordon) [inline lib/rom/repository.rb:164]: I guess you mean `container.mappers[relation.name][custom_mapper]`. I mean mappers registered for a relation are scoped by relation name so we have to "unscope" them first

**Comment by flash-gordon** [comment]
Yep, sure

**Comment by solnic** [comment]
@flash-gordon you have commit access btw

**Comment by flash-gordon** [comment]
Yeah, thanks. Github sent me an email as well :)

File: lib/rom/repository.rb (hunk 1 above)
Comment (solnic) [inline lib/rom/repository.rb:164]: Uhm, yes.

**Comment by flash-gordon** [comment]
@solnic done. Thanks for the feedback, I don't like inversed conditions either ;)

**Pull request merged**